pkg/tools/etcd_helper.go

```diff
@@ -284,31 +284,33 @@ func (h *EtcdHelper) SetObj(key string, obj, out runtime.Object, ttl uint64) err
 	return err
 }
 
-// Pass an EtcdUpdateFunc to EtcdHelper.AtomicUpdate to make an atomic etcd update.
-// See the comment for AtomicUpdate for more detail.
+// Pass an EtcdUpdateFunc to EtcdHelper.GuaranteedUpdate to make an etcd update that is guaranteed to succeed.
+// See the comment for GuaranteedUpdate for more detail.
 type EtcdUpdateFunc func(input runtime.Object) (output runtime.Object, ttl uint64, err error)
 
-// AtomicUpdate generalizes the pattern that allows for making atomic updates to etcd objects.
-// Note, tryUpdate may be called more than once.
+// GuaranteedUpdate calls "tryUpdate()" to update key "key" that is of type "ptrToType". It keeps
+// retrying the update even in the face of etcd index conflict, until it succeeds. Note that object
```

Comment:
It retries only if  "IsEtcdTestFailed" returns true - IIUC this is the case only if there is precondition failure. So maybe it's worth mentioning that it retires only if there is precondition failure (or index conflict which I understood as exactly the same) and doesn't retry on any other errors?